sky/backends/cloud_vm_ray_backend.py

```diff
@@ -2769,10 +2769,8 @@ def check_resources_fit_cluster(
         requested_resource_list = []
         for resource in task.resources:
             if (task.num_nodes <= handle.launched_nodes and
-                    resource.less_demanding_than(
-                        launched_resources,
-                        requested_num_nodes=task.num_nodes,
```

Comment:
Why do we remove this parameter? I can see that we check the node count on L2771, but it still seems useful to keep it here as well.